File: ducks/MiniDuckSimulator.java
public class MiniDuckSimulator {
  public static void main(String[] args) {
    Duck mallard = new MallardDuck();
    mallard.performQuack();
    mallard.performFly();
    Duck model = new ModelDuck();
    model.performFly();
    model.setFlyBehavior(new FlyRocketPowered());
    model.performFly();
    Duck myduck = new MyDuck();
    myduck.display();
    myduck.swim();
    myduck.performQuack();
  }
}

File: php/strategy/MiniDuckSimulator.php
<?php
namespace ducks;
require('MallardDuck.php');
require('ModelDuck.php');

use ducks\MallardDuck;


$mallard = new MallardDuck();
$mallard->performQuack();
$mallard->performFly();
$mallard->display();
$mallard->swim();
$model = new ModelDuck();
$model->performFly();
$model->performQuack();


